### DISCUSSION TITLE
Plans to add maxPitch > 60 support?

### DISCUSSION THREAD

**Discussion by mfitzer** [maplibre/maplibre-gl-js#763] [Q&A]
I'm working on a project that requires the map to be tilted at a pitch of more than 60 degrees. Are there plans to support this behavior?

**Comment by acalcutt** [comment]
I could be wrong, but I think right now you can go up to 85 degrees in the current 2.0 pre release if you specify maxPitch in your map options. It was part of this PR ( https://github.com/maplibre/maplibre-gl-js/pull/574 )

```
		var map = new maplibregl.Map({
			container: 'map',
			style: 'https://tile.url/styles/style/style.json',
			center: [-95.009766,37.09024],
			zoom: 2,
			pitch: 0,
			bearing: 0,
			maxPitch: 85,
		});
```

I currently use this in my wifidb.net map along with the css to make a gradient background they mention in ( https://github.com/maplibre/maplibre-gl-js/pull/574#issuecomment-977017379 )

#map{
    background: linear-gradient(0deg, rgba(36,0,0,1) 56%, rgb(249 209 89) 62%, rgb(81 151 173) 80%, rgb(2 92 179) 95%)
}

**Reply by mfitzer** [reply]
@acalcutt Good to know! Do you know if there are plans to support even larger pitch values ranging all the way to 180?

**Reply by acalcutt** [reply]
I am not directly involved with this project, so I only know what I have read here and tried myself. However in that PR and other things I have seen it sounds like above 85 has a few issues that need to be worked out. 85 was considered a starting point.

I am using my own fork that includes the test terrain features, so in my case it may be a little different. In the 3d terrain, the 85 degree change helps a lot and works good on newer hardware. it can get a bit crashy on older hardware.  In the [terrain 3d branch](https://github.com/prozessor13/maplibre-gl-js/tree/terrain3d) I am merging in, some of these issues could have been addressed that would help, but I don't know for sure.

**Reply by acalcutt** [reply]
I do know in the 3d terrain branch they have started trying to make the camera stay at the same altitude when the pitch is changed, which I think would help the >85 pitch. but don't take my word for it. Whenever I read those camera code changes it makes my head hurt.

**Reply by HarelM** [reply]
Currently the implementation of pitch > 60 is "at your own risk", meaning that it may load too many tiles near the horizon.
It would be great if someone from the community will pick this up and solves this properly. :-) until then it's in "experimental" mode I guess...

**Reply by mfitzer** [reply]
@acalcutt @HarelM Thank you for the information, that is very helpful!

**Reply by clement-igonet** [reply]
This could help for quick test of maxPitch at any value:
```javascript
 map.setMaxPitch = (maxPitch) => {
             map.transform.maxPitch = maxPitch;
             map._update();
             return map;
 }
```
